benches/large_case.rs

```diff
@@ -1,22 +1,38 @@
 // SPDX-License-Identifier: MPL-2.0
+use std::time::Duration;
 
-#![feature(test)]
-extern crate test;
-use test::Bencher;
+extern crate criterion;
+use self::criterion::*;
 
-use pubgrub::solver::{resolve, OfflineDependencyProvider};
+use pubgrub::solver::{resolve, DependencyProvider, OfflineDependencyProvider};
 use pubgrub::version::NumberVersion;
 
-#[cfg(feature = "serde")]
-#[bench]
-/// This is an entirely synthetic benchmark. It may not be realistic.
-fn large_case(b: &mut Bencher) {
-    let s = std::fs::read_to_string("test-examples/large_case_u16_NumberVersion.ron").unwrap();
-    let dependency_provider: OfflineDependencyProvider<u16, NumberVersion> =
-        ron::de::from_str(&s).unwrap();
-
-    // bench
-    b.iter(|| {
-        let _ = resolve(&dependency_provider, 0, 0);
-    });
+fn bench_nested(c: &mut Criterion) {
+    let mut group = c.benchmark_group("large_cases");
+    group.measurement_time(Duration::from_secs(10));
```

Comment:
If I assume correctly, the time here is very machine dependent. Criterion suggests adding a couple seconds here on my laptop. It would be best if we all were able to run benchmarks without changes.

Are there any other options we could use instead of the time? Hard to suggest anything concrete without diving into the API myself, will do so later.

When I looked through the API, I saw `group.confidence_interval` (or similar), where you set up a desirable level. Maybe that or some other option would do the trick.

Reply:
It is also very dependent on optimizations, as the code gets faster the time can go down. I picked 10 working on top of a bunch of wip perf commits. I will change it to 20 and we can adjust as needed.

Unfortunately, `confidence_interval` does not affect the measurement time only the amount of difference that is declared `regressed`.

Reply:
Thank you for the change :+1: 

> Unfortunately, confidence_interval does not affect the measurement time only the amount of difference that is declared regressed.

Yeah, that's unfortunate. Since we don't know any better options other than time, let's go with that.